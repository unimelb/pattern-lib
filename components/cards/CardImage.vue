<template>
  <a
    :href="href"
    :aria-label="title"
    class="btn-owner card card--image card--bdr bg-inverted">
    <div
      v-if="thumb && thumb.length"
      :style="{ backgroundImage: `url(${thumb})` }"
      :aria-label="title"
      class="card__thumb card__thumb--zoom" />
    <div class="card__inner">
      <ButtonLink
        class="btn--inverted btn--fullwidth"
        element="span">{{ title }}</ButtonLink>
      <p>
        <slot />
      </p>
    </div>
  </a>
</template>

<script>
import ButtonLink from '../buttons/ButtonLink.vue';

export default {
  name: 'CardImage',
  components: {
    ButtonLink,
  },
  props: {
    thumb: {
      type: String,
      default: '',
    },
    title: {
      type: String,
      default: 'Test title',
    },
    href: {
      type: String,
      default: '',
      required: true,
    },
  },
};
</script>
