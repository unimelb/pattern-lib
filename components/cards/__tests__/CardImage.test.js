import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import CardImage from '../CardImage.vue';

expect.extend(toHaveNoViolations);

describe('CardImage', () => {
  it('should match snapshot', () => {
    const result = shallow(CardImage, {
      propsData: {
        title: 'mock title',
        href: 'www.unimelb.edu.au',
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(CardImage, {
      propsData: {
        title: 'mock title',
        href: 'www.unimelb.edu.au',
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
