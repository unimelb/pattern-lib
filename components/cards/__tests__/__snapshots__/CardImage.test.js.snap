// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImage should match snapshot 1`] = `
<a
  aria-label="mock title"
  class="btn-owner card card--image card--bdr bg-inverted"
  href="www.unimelb.edu.au"
>
  <!---->
   
  <div
    class="card__inner"
  >
    <!---->
     
    <p />
  </div>
</a>
`;
